### ISSUE TITLE
[BUG] On re-authentiction the currently logged in user is ignored

### ISSUE THREAD

**Issue by BlackVoid** [sventorben/keycloak-home-idp-discovery#309]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Alt1.
1. Login to keycloak wth any application
2. Wait until a re-authentication is required (not sure about exact time but about 5 minutes)
3. Trigger a login from a app with a action included ex. `kc_action=UPDATE_PROFILE`
4. Form is shown with attempted user being the user I'm logged in as and email field is empty
5. User has to enter the username or select the IDP
6. Eventually user gets to the update profile form

![image](https://github.com/sventorben/keycloak-home-idp-discovery/assets/673720/c7428250-a7c1-4ea4-8aa7-2181c5912fc3)

Alt2.
1. Login to keycloak wth any application
2. Wait until a re-authentication is required (not sure about exact time but about 5 minutes)
3. Trigger a login from a app with a action included ex. `kc_action=UPDATE_PROFILE` and include `login_hint=myemail`
4. Get redirected to 3rd party IDP and then redirected back to Keycloak
7. Update profile form is shown

### Expected Behavior

Behavior of Alt1 should be that the user that of Alt2 after step 3 i.e:
1. Login to keycloak wth any application
2. Wait until a re-authentication is required (not sure about exact time but about 5 minutes)
3. Trigger a login from a app with a action included ex. `kc_action=UPDATE_PROFILE`
4. Get redirected to 3rd party IDP and then redirected back to Keycloak
7. Update profile form is shown

If I don't have a IDP connected to the user and it only has a password it also works as expected, see the following image

![image](https://github.com/sventorben/keycloak-home-idp-discovery/assets/673720/49de2547-901f-439e-b663-a2444b4b525c)

### Steps To Reproduce

Configure the browser flow to contain the following steps:
   - Cookie - Alternative
   - Home IdP Discovery - Alternative
      - User attribute: email
      - Forward users with unverified email: true
      - Bypass login page: true
      - Forward to linked IdP: true
      - Forward to first matched IdP: false
   - Password Form (used during re-authentiction) - Alternative
   - Username Password Form (used during login) - Alternative

Alt1.
1. With a user connected to any 3rd party IdP
2. Login to keycloak wth any application
3. Wait until a re-authentication is required (not sure about exact time but about 5 minutes)
5. Trigger a login from a app with a action included ex. `kc_action=UPDATE_PROFILE`
6. Form is shown with attempted user being the user I'm logged in as and email field is empty
7. User has to enter the username or select the IDP
8. Eventually user gets to the update profile form

![image](https://github.com/sventorben/keycloak-home-idp-discovery/assets/673720/c7428250-a7c1-4ea4-8aa7-2181c5912fc3)

Alt2.
1. With a user connected to any 3rd party IdP
2. Login to keycloak wth any application
3. Wait until a re-authentication is required (not sure about exact time but about 5 minutes)
4. Trigger a login from a app with a action included ex. `kc_action=UPDATE_PROFILE` and include `login_hint=myemail`
5. Get redirected to 3rd party IDP and then redirected back to Keycloak
6. Update profile form is shown

### Version

```markdown
- Keycloak: 23.0.4
- This extension: 23.0.0
```

**Comment by robson90** [comment]
@BlackVoid Im very sorry, but could please elaborate, why you configured you flow lioke you did ? 
And do you have a clear example for the use case?

**Comment by BlackVoid** [comment]
Okay, so from the beginning I had the following flow, just as the documentation suggests:

- Cookie - Alternative
- Home IdP Discovery - Alternative
- Username Password Form - Alternative

What I discovered is when re-authentication [1] is required by keycloak, such as when you [trigger an action](https://github.com/keycloak/keycloak-community/blob/main/design/application-initiated-actions.md) on the user from any client, then the user has to enter the username again even though the user is already signed which does not make a lot of sense.

While playing around I discovered that if I add the password form before the username-password form, then keycloak will skip it during logins, but not during re-authentication attempts, so the user ends up only entering the password if they don't have a IdP connected to their account.

Back to the reported issue it still persists with the documented flow, so it makes me think the desired logic is only triggered by `login_hint` and not if the user is already signed in.

[1] re-authentication is only required after a certain time after login
